src/Main.js

```diff
@@ -0,0 +1,214 @@
+import React, { useState } from 'react';
+import { useLocation, useHistory } from 'react-router-dom';
+import PeopleTable from './PeopleTable';
+
+const url = 'https://mate-academy.github.io/react_people-table/api/people.json';
+
+const Main = () => {
+  const [peopleAmount, setPeopleAmount] = useState(0);
+  const [isLoading, setIsLoading] = useState(true);
+  const [people, setPeople] = useState([]);
+  const [peopleList, setPeopleList] = useState([]);
+  const [query, setQuery] = useState(null);
+  const [sortBy, setSortBy] = useState(null);
+
+  const location = useLocation();
+  const history = useHistory();
+  const searchParams = new URLSearchParams(location.search);
+  const sortTags = ['name', 'id', 'sex', 'born', 'died', 'age', 'century'];
+  const nameFromUrl = searchParams.get('query');
+  const sortByFromUrl = searchParams.get('sortBy');
+
+  if (nameFromUrl && nameFromUrl !== query) {
+    setQuery(nameFromUrl);
+    findPerson(nameFromUrl);
+  }
+
+  if (sortByFromUrl && sortByFromUrl !== sortBy) {
+    setSortBy(sortByFromUrl);
+    sort(sortByFromUrl);
+  }
+
+  if (isLoading) {
+    fetch(url).then(listOfPeople => listOfPeople.json())
+      .then((list) => {
+        setPeopleAmount(list.length);
+        setPeople(list.map((person, index) => ({
+          ...person,
+          id: index,
+        })));
+        setPeopleList(list.map((person, index) => ({
+          ...person,
+          id: index,
+        })));
+        setIsLoading(false);
+      });
+  }
+
+  const debounce = (f, delay) => {
+    let timer;
+
+    return (...args) => {
+      clearTimeout(timer);
+      timer = setTimeout(() => {
+        f(...args);
+      }, delay);
+    };
+  };
+
+  function findPerson(name) {
+    const filteredPeople = people.filter((person) => {
+      const mother = person.mother === null ? '' : person.mother;
+      const father = person.father === null ? '' : person.father;
+
+      return (
+        mother.toLowerCase().includes(name.toLowerCase())
+        || person.name.toLowerCase().includes(name.toLowerCase())
+        || father.toLowerCase().includes(name.toLowerCase())
+      );
+    });
+
+    setPeopleList(filteredPeople);
+    setPeopleAmount(filteredPeople.length);
+
+    if (name.length > 0) {
+      searchParams.set('query', name);
+      history.push({ search: searchParams.toString() });
+    } else {
+      history.push({ search: null });
+    }
+  }
+
+  const inputText = debounce(findPerson, 1000);
+
+  function sort(e) {
+    let sortedPeople = [];
+    const sortName = typeof (e) === 'string' ? e : e.target.value;
+    const previousOrder = [...peopleList];
+
+    switch (sortName) {
```

Comment:
```suggestion
    switch (typeof previouseOrder[0][sortName]) {
```
Тогда тебе нужно будет проверять всего два кейса: 'string' и 'number'